akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -154,15 +154,31 @@ private[akka] final class LogMessagesInterceptor(val opts: LogOptions) extends B
 
   import BehaviorInterceptor._
 
+  val log = LoggerFactory.getLogger(classOf[BehaviorInterceptor[Any, Any]])
+
   override def aroundReceive(ctx: TypedActorContext[Any], msg: Any, target: ReceiveTarget[Any]): Behavior[Any] = {
     if (opts.enabled)
-      opts.logger.getOrElse(ctx.asScala.log).log(opts.level, "received message {}", msg)
+      opts.level match {
+        case Logging.ErrorLevel   => log.error("received message {}", msg)
+        case Logging.WarningLevel => log.warn("received message {}", msg)
+        case Logging.InfoLevel    => log.info("received message {}", msg)
+        case Logging.DebugLevel   => log.debug("received message {}", msg)
```

Comment:
The type of  `opts.level` should be the native Slf4J `org.slf4j.event.Level` instead now.

Reply:
That's a shortcoming to not be able to log at a given level with a parameter. Probably most annoying for us as a library and not so much for end users.

Reply:
you right `org.slf4j.event.Level` is the proper one to use.